src/components/Settings/SettingsProfile.js

```diff
@@ -0,0 +1,68 @@
+import {Button, Image, Text, TextInput, View} from "react-native";
+import {viewStyles} from "../../styles/settings/settings";
+import {launchImageLibrary} from "react-native-image-picker";
+import React from "react";
+
+const SettingsProfile = ( { settings, onSettingsModified } ): React.Node => {
+  const profileSource =
+    settings.removeProfilePhoto || !settings.icon ?
+      require( "./../../images/profile.png" ) :
+      { uri: settings.newProfilePhoto ? settings.newProfilePhoto.uri : settings.icon};
+
+  const onImageSelected = ( response ) => {
+    if ( response.didCancel ) {return;}
+    onSettingsModified( { ...settings, newProfilePhoto: response.assets[0], removeProfilePhoto: false } );
```

Comment:
Can we display the new profile photo after a user selects it from `launchImageLibrary`? Otherwise it's hard to know whether the selection actually worked.

Reply:
Weird, it's working for me on Android - still not showing anything after you select an image on iOS?

Reply:
This doesn't work for me in iOS or in Android, as in I don't see the image updated in the app, and the request to stagingapi fails... I think b/c PUT /v2/users/:id is [documented but not implemented](https://github.com/inaturalist/iNaturalistAPI/issues/323). Yaron, while we're working on that, can we get the image to appear even before it uploads so the user can see that something changed?

Reply:
@kueda - Very odd, I just tested the preview-before-upload on Android and it did work for me (meaning, choosing "Upload new photo" then the selected photo gets displayed in the app). It doesn't show the already-uploaded photo, since the v2 staging API shows a URL in the `staticdev.inaturalist.org` domain, which returns S3 error of access denied.
Maybe it's a permissions issue?

Reply:
Not sure why it's not working for you, but for me, `const profileSource` is not getting set when `settings.icon` is blank and you choose a new and you choose a new profile photo without removing the old one first. This change works for me:

```javascript
let profileSource;
if ( settings.newProfilePhoto ) { profileSource = { uri: settings.newProfilePhoto.uri }; }
else if ( settings.icon ) { profileSource = { uri: settings.icon }; }
else { profileSource = require( "./../../images/profile.png" ); }
```

Reply:
Used your latest code with some modifications, since it didn't work if you chose a photo then removed it